index.html

```diff
@@ -1198,23 +1202,40 @@
         NDEFRecordData data;
       };
     </pre>
+    <p>
+      <a>NDEFRecord</a> objects have an associated <dfn id="ndefrecord-bytes">bytes</dfn>
+      (a <a>byte sequence</a>) set on creation that represents the payload data of the
+      <a>NDEF record</a>, which is `null` if there was no data in the
+      <a>NDEF record</a>.
+    </p>
     <p>
       The <dfn>NDEFRecordData</dfn> is a union type representing data types allowed
       for <a href="#dom-ndefrecordinit-data">NDEFRecordInit.data</a> property.
     </p>
     <p>
-      The <dfn>NDEFRecord.mediaType</dfn>
+      The <dfn data-dfn-for="NDEFRecord">mediaType</dfn>
       property represents the <a>MIME type</a> of the <a>NDEF record</a>
       payload.
     </p>
     <p>
-      The <dfn>NDEFRecord.recordType</dfn> property represents the
+      The <dfn data-dfn-for="NDEFRecord">recordType</dfn> property represents the
       <a>NDEF record</a> types.
     </p>
     <p>
-      The <dfn>NDEFRecord.data()</dfn> method returns the
-      payload data of the <a>NDEF record</a> with an appropriate [[!ECMAScript]]
-      type, which depends on the <a>MIME type</a>.
+      The <dfn data-dfn-for="NDEFRecord">toText()</dfn> method, when invoked, MUST return the
+      result of running <a>convert NDEFRecord's bytes</a> with an <a>NDEFRecord</a> object
+      and a `text` type, unless there's other specific illustration in step 7 of
```

Comment:
I don't understand "there's other specific illustration in step 7" 

other makes it sound like plural, so maybe you want to say "another"? but what is specific illustration in ? Also @marcoscaceres can we somehow refer to that instead of hardcoding 7 here?

Reply:
I meant this step: 
> Otherwise, set the result of invoking record's toText() method to the result of running decode on buffer with encoding set to "utf-16be".

Which is specified for `utf-16be` decode. I don't have better idea, so help need. :)

Reply:
So what you are referring to is 

              <li>
                If <var>charset</var> is equal to <code>"utf-8"</code>,
                set <var>record</var>'s data to the result of running
                <a>UTF-8 decode</a> on <var>buffer</var>.
                set <var>record</var>'s associated <a>bytes</a> to <var>buffer</var>.
              </li>
              <li>
                Otherwise, set <var>record</var>'s data to the result of running
                Otherwise, set the result of invoking <var>record</var>'s
                <a href="#dom-ndefrecord-totext">toText()</a> method to the result of running
                <a>decode</a> on <var>buffer</var> with <var>encoding</var> set
                to <code>"utf-16be"</code>.
              </li>

Reply:
Exactly.

Reply:
Sounds to me that the "convert NDEFRecord's bytes" algorithm should handle the case that the data is encoded as UTF-16

Reply:
I understand the issue now. We loose info about the charset in the reading step.

Maybe we could do

> 2. Let charset be "utf-8" if bit 7 (most significant bit) of header is equal to the value 0, or else "utf-16be". 

> 3, Let MIME type's parameters["charset"] be charset, in the case it is not equal to "utf-8"

Reply:
The header is the first byte of ndef's PAYLOAD field, and we didn't pass the header to the record's bytes. So we don't know what the charset is.

Maybe we could set the record's bytes to the whole byte sequence of ndef's PAYLOAD field if the recordType is 'text', then we could calculate the charset. Does that make sense?

Reply:
I think it does

Reply:
Done, PTAL again. Thanks!:)